drivers/platform/pico/pico_gpio.c

```diff
@@ -0,0 +1,290 @@
+ * @param param - GPIO initialization parameters
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t pico_gpio_get(struct no_os_gpio_desc **desc,
+		      const struct no_os_gpio_init_param *param)
+{
+	struct no_os_gpio_desc *descriptor;
+	struct pico_gpio_desc *extra;
+	int32_t ret;
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	descriptor = (struct no_os_gpio_desc *)malloc(sizeof(*descriptor));
+	if (!descriptor) {
+		ret = -ENOMEM;
+		goto error;
+	}
+
+	extra = (struct pico_gpio_desc*)malloc(sizeof(*extra));
+	if (!extra) {
+		ret = -ENOMEM;
+		goto error;
+	}
+
+	descriptor->extra = extra;
+	ret = _gpio_init(descriptor, param);
+	if(ret)
+		goto error;
+
+	*desc = descriptor;
+
+	return 0;
+error:
+	free(extra);
+	free(descriptor);
+
+	return ret;
+}
+
+/**
+ * @brief Get the value of an optional GPIO.
+ * @param desc  - The GPIO descriptor.
+ * @param param - GPIO Initialization parameters.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t pico_gpio_get_optional(struct no_os_gpio_desc **desc,
+			       const struct no_os_gpio_init_param *param)
+{
+	if(param == NULL) {
+		*desc = NULL;
+		return 0;
+	}
+
+	return pico_gpio_get(desc, param);
+}
+
+/**
+ * @brief Free the resources allocated by no_os_gpio_get().
+ * @param desc - The GPIO descriptor.
+ * @return 0.
+ */
+int32_t pico_gpio_remove(struct no_os_gpio_desc *desc)
+{
+	if (desc != NULL)
+		free(desc->extra);
+
+	free(desc);
+
+	return 0;
+}
+
+/**
+ * @brief Enable the input direction of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t pico_gpio_direction_input(struct no_os_gpio_desc *desc)
+{
+	struct pico_gpio_desc *pico_gpio;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	pico_gpio = desc->extra;
+
+	gpio_set_input_enabled(desc->number, true);
+	pico_gpio->input_enabled = true;
+
+	return 0;
+}
+
+/**
+ * @brief Enable the output direction of the specified GPIO.
+ * @param desc  - The GPIO descriptor.
+ * @param value - The value.
+ *                Example: NO_OS_GPIO_HIGH
+ *                         NO_OS_GPIO_LOW
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t pico_gpio_direction_output(struct no_os_gpio_desc *desc,
+				   uint8_t value)
+{
+	struct pico_gpio_desc *pico_gpio;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	pico_gpio = desc->extra;
+
+	gpio_set_input_enabled(desc->number, false);
+	pico_gpio->input_enabled = false;
+
+	return 0;
+}
+
+/**
+ * @brief Get the direction of the specified GPIO.
+ * @param desc      - The GPIO descriptor.
+ * @param direction - The direction.
+ *                    Example: NO_OS_GPIO_OUT
+ *                             NO_OS_GPIO_IN
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t pico_gpio_get_direction(struct no_os_gpio_desc *desc,
+				uint8_t *direction)
+{
+	struct pico_gpio_desc *pico_gpio;
+
+	if (!desc || !desc->extra || !direction)
```

Comment:
seeing how we write to that memory I think it is nice to have just in case the user forget to add anything in the direction parameter when calling the function